packages/qwik/src/core/v2/shared/component-execution.ts

```diff
@@ -153,6 +179,10 @@ function findFirstStringJSX(jsx: JSXOutput): JSXNode<string> | null {
       queue.push(jsx.children);
     } else if (Array.isArray(jsx)) {
       queue.push(...jsx);
+    } else if (isPromise(jsx)) {
+      return maybeThen<JSXOutput, JSXNode<string> | null>(jsx, (jsx) => findFirstStringJSX(jsx));
+    } else if (isSignal(jsx)) {
+      return findFirstStringJSX(untrack(() => jsx.value as JSXOutput));
```

Comment:
This is the important change. We could get the jsx from a promise or a signal value, so we need to check if the returned value is a JSXNode.